kf/kf_unix.go

```diff
@@ -247,45 +289,67 @@ func (b *BPF) LoadTCAttachProgram(ifaceName, direction string) error {
 				Kind: "clsact",
 			},
 		}
-
 		if err := tcgo.Qdisc().Add(&qdisc); err != nil {
 			log.Info().Msgf("could not assign clsact to %s : %v, its already exists", ifaceName, err)
 		}
-	}
 
-	bpfRootProg := b.ProgMapCollection.Programs[b.Program.EntryFunctionName]
+		if direction == models.IngressType {
+			parent = tc.HandleMinIngress
+		} else if direction == models.EgressType {
+			parent = tc.HandleMinEgress
+		}
+		progFD := uint32(bpfRootProg.FD())
+		// Netlink attribute used in the Linux kernel
+		bpfFlag := uint32(tc.BpfActDirect)
 
-	var parent uint32
-	if direction == models.IngressType {
-		parent = tc.HandleMinIngress
-	} else if direction == models.EgressType {
-		parent = tc.HandleMinEgress
-	}
+		filter = tc.Object{
+			Msg: tc.Msg{
+				Family:  unix.AF_UNSPEC,
+				Ifindex: uint32(iface.Index),
+				Handle:  0,
+				Parent:  core.BuildHandle(tc.HandleRoot, parent),
+				Info:    0x300,
+			},
+			Attribute: tc.Attribute{
+				Kind: "bpf",
+				BPF: &tc.Bpf{
+					FD:    &progFD,
+					Flags: &bpfFlag,
+				},
+			},
```

Comment:
Is it possible to define tc.Attribute{} outside of the block in order to prevent duplicated code?

Reply:
As tc.Attribute{} is the field of tc.Object.Attribute, I would prefer to keep the declaration in place for context.